devito/passes/iet/openmp.py

```diff
@@ -135,8 +136,22 @@ def _make_clauses(cls, ncollapse=None, chunk_size=None, nthreads=None,
             clauses.append('num_threads(%s)' % nthreads)
 
         if reduction:
-            args = ','.join(str(i) for i in reduction)
-            clauses.append('reduction(+:%s)' % args)
+            args = []
+            for i in reduction:
+                if i.is_Indexed:
+                    # OpenMP expects a size not an index as input of reduction,
+                    # such as reduction(+:f[d_m:d_M+1])
```

Comment:
Need to update this since not the generated anymore.

Reply:
gotcha, fixed